### PR TITLE
AccessControl: Remove legacy frontend AC checks

### PR DESCRIPTION
**Pull request by Jguer**
AccessControl: Remove legacy frontend AC checks

https://github.com/grafana/grafana-enterprise/pull/5873

### CODE REVIEW

File: public/app/core/services/context_srv.ts [hunk 1]
```diff
@@ -180,23 +164,14 @@ export class ContextSrv {
...
 
   hasAccessToExplore() {
-    if (this.accessControlEnabled()) {
-      return this.hasPermission(AccessControlAction.DataSourcesExplore) && config.exploreEnabled;
-    }
-    return (this.isEditor || config.viewersCanEdit) && config.exploreEnabled;
+    return this.hasPermission(AccessControlAction.DataSourcesExplore) && config.exploreEnabled;
   }
 
   hasAccess(action: string, fallBack: boolean): boolean {
```
Comment (IevaVasiljeva) [inline public/app/core/services/context_srv.ts:170]: Can you please make a note to remove this and `hasAccessInMetadata` altogether (should be replaced by `hasPermission` and `hasPermissionInMetadata`)?

File: public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx [hunk 2]
```diff
@@ -148,14 +147,7 @@ function getSettingsPages(dashboard: DashboardModel) {
   const permissionsTitle = t('dashboard-settings.permissions.title', 'Permissions');
 
   if (dashboard.id && dashboard.meta.canAdmin) {
-    if (!config.rbacEnabled) {
-      pages.push({
-        title: permissionsTitle,
-        id: 'permissions',
-        icon: 'lock',
-        component: DashboardPermissions,
```
Comment (IevaVasiljeva) [inline public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx:156]: I think we can remove `DashboardPermissions` altogether.

**Review by IevaVasiljeva — APPROVED** [review]

File: public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx (hunk 2 above)
Comment (Jguer) [reply public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx:156]: :+1: will remove it in the next PR

File: public/app/core/services/context_srv.ts (hunk 1 above)
Comment (Jguer) [reply public/app/core/services/context_srv.ts:170]: tracked in https://github.com/grafana/grafana/issues/74384

File: public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx (hunk 2 above)
Comment (Jguer) [reply public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx:156]: tracked in https://github.com/grafana/grafana/issues/76203

**Pull request merged**